module-4/backend/week_one.md

```diff
@@ -5,15 +5,52 @@ Fork this respository. Answer the questions to the best of your ability. Try to
 Note: When you're done, submit a PR. 
 
 1. What's the most useful thing you learned from completing the intermission week work?
+
+`Understanding the basics of Javascript and how it differs from the syntax of Ruby.` 
+
 2. What are some tools to help debug JavaScript code?
+
+ `Console.log, Debugger.`
+ 
 3. What are some tools you need in order to unit test your JavaScript?
+
+`Mocha, Chai.`
+
 4. What is the syntax for invoking a function?
-5. What's `this` in JavaScript?
-6. What is Webpack and why is it useful?
-7. When/why do you want to use event delegation?
-8. What's `npm` and what do we use it for?
+
+`function myFunction(x,y) {
+  return x + y;
+  }
+myFunction(300,1);`
+
+5. What's the difference between `==` and `===` in JavaScript?
+
+`There is no type conversion with the `===`.` 
+
+6. What's the difference between asynchronous and synchronous JavaScript? 
+
+`Synchronous waits for each operation to complete while asynchronous does not. Synchronous will not run the next operation if the result of the previous is not complete. Asynchronous will run all operations at once.`  
+
+7. How do we setup a route when creating an API with Node and Express?
+
+8. What do we use Knex for?
+
+`Knex is a SQL query builder. It allows us to interact with our db through written queries.` 
+
+9. What's `npm` and what do we use it for?
+
+`Its a package manager. We use it to install plugins that help us to write code.` 
 
 #### Review  
-9. What's the MVC design pattern? Describe each part of MVC.
-10. What are a few ways to optimize a Rails application?
-11. What's a background worker? When would we want to use a background worker?
+10. What's the MVC design pattern? Describe each part of MVC?
+
+`User makes a request to the controller which talks to the model. The model returns the info to the controller and renders the proper view.`
```

Comment:
this is a pretty common interview question - i suggest you add more detail and practice this question!